storm-core/src/clj/backtype/storm/daemon/worker.clj

```diff
@@ -510,15 +554,21 @@
                  ))
              )
         credentials (atom initial-credentials)
-        check-credentials-changed (fn []
-                                    (let [new-creds (.credentials (:storm-cluster-state worker) storm-id nil)]
-                                      (when-not (= new-creds @credentials) ;;This does not have to be atomic, worst case we update when one is not needed
-                                        (AuthUtils/updateSubject subject auto-creds new-creds)
-                                        (dofor [e @executors] (.credentials-changed e new-creds))
-                                        (reset! credentials new-creds))))
-      ]
-    (.credentials (:storm-cluster-state worker) storm-id (fn [args] (check-credentials-changed)))
-    (schedule-recurring (:refresh-credentials-timer worker) 0 (conf TASK-CREDENTIALS-POLL-SECS) check-credentials-changed)
+        check-credentials-throttle-changed (fn []
+                                             (let [callback (fn cb [& ignored] 
+                                                              (let [throttle-on (.topology-backpressure (:storm-cluster-state worker) storm-id cb)]
+                                                                (reset! (:throttle-on worker) throttle-on)))
+                                                   new-throttle-on (if ((:storm-conf worker) TOPOLOGY-BACKPRESSURE-ENABLE)
+                                                                     (.topology-backpressure (:storm-cluster-state worker) storm-id callback) nil)
+                                                   new-creds (.credentials (:storm-cluster-state worker) storm-id nil)]
+                                               (if ((:storm-conf worker) TOPOLOGY-BACKPRESSURE-ENABLE)
+                                                 (reset! (:throttle-on worker) new-throttle-on))
+                                               (when-not (= new-creds @credentials) ;;This does not have to be atomic, worst case we update when one is not needed
+                                                 (AuthUtils/updateSubject subject auto-creds new-creds)
+                                                 (dofor [e @executors] (.credentials-changed e new-creds))
+                                                 (reset! credentials new-creds))))]
+    (.credentials (:storm-cluster-state worker) storm-id (fn [args] (check-credentials-throttle-changed)))
```

Comment:
I think I would prefer to have two functions instead of just one.  Both can be scheduled on the same timer, but keeping them separate seems cleaner.